bee-inx/src/block.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use bee_block as bee;
+use inx::proto;
+
+/// The [`Block`] type.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Block {
+    /// The [`BlockId`](bee::BlockId) of the block.
+    pub block_id: bee::BlockId,
+    /// The complete [`Block`](bee::Block).
+    pub block: bee::Block,
+    /// The raw bytes of the block.
+    pub raw: Vec<u8>,
+}
+
+/// The [`BlockWithMetadata`] type.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct BlockWithMetadata {
+    /// The [`Metadata`](crate::BlockMetadata) of the block.
+    pub metadata: crate::BlockMetadata,
+    /// The complete [`Block`](bee::Block).
+    pub block: bee::Block,
+    /// The raw bytes of the block.
+    pub raw: Vec<u8>,
+}
+
+impl TryFrom<proto::BlockWithMetadata> for BlockWithMetadata {
+    type Error = bee::InxError;
+
+    fn try_from(value: proto::BlockWithMetadata) -> Result<Self, Self::Error> {
+        let raw = value.block.ok_or(bee::InxError::MissingField("block"))?;
+        Ok(BlockWithMetadata {
+            metadata: value
+                .metadata
+                .ok_or(Self::Error::MissingField("metadata"))?
+                .try_into()?,
+            block: raw.clone().try_into()?,
+            raw: raw.data,
```

Comment:
It's probably(!) more performant to do a mem-copy of `raw.data` beforehand into a `data` Vec, and then just call `try_into` on the `raw` struct itself.

Reply:
I should have actually just used the the `Raw<bee::Block>` struct that I created for these cases. Now the user can choose when the copy is done and the performance hit occurs.